.github/workflows/reusable-native-tests.yml

```diff
@@ -21,7 +21,7 @@ jobs:
       matrix:
         test-java-version:
           - 22
-          - 23
+          - 24
```

Comment:
do we want to say we support `23`? if so, I'd suggest that we keep `23` in the matrix

Reply:
I added 23 again but it will make the build longer.

Reply:
I don't think it shouldn't make the build longer (they matrixed jobs run in parallel)

Reply:
Matrixed jobs run in parallel but the available ressources (CPU, memory) are not infinite. Native compilation can consume significant resources.